src/Sentry.AspNetCore/SentryAspNetCoreOptionsSetup.cs

```diff
@@ -25,10 +25,52 @@ public override void Configure(SentryAspNetCoreOptions options)
         {
             base.Configure(options);
 
-            // Don't override user defined value
-            options.Environment ??=
-                EnvironmentLocator.Locate() // Sentry specific environment takes precedence #92
-                ?? _hostingEnvironment.EnvironmentName;
+            // Don't override user defined value.
+            if (string.IsNullOrWhiteSpace(options.Environment))
+            {
+                var locatedEnvironment = EnvironmentLocator.Locate();
+                if (!string.IsNullOrWhiteSpace(locatedEnvironment))
+                {
+                    // Sentry specific environment takes precedence #92.
+                    options.Environment = locatedEnvironment;
+                }
+                else
+                {
+                    // NOTE: Sentry prefers to have it's environment setting to be all lower case.
+                    //       .NET Core sets the ENV variable to 'Production' (upper case P) or
+                    //       'Development' (upper case D) which conflicts with the Sentry recommendation.
+                    //       As such, we'll be kind and override those values, here ... if applicable.
+                    // Assumption: The Hosting Environment is always set.
+                    //             If not set by a developer, then the framework will auto set it.
+                    //             Alternatively, developers might set this to a CUSTOM value, which we
+                    //             need to respect (especially the case-sensitivity).
+                    //             REF: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments
+#if NETSTANDARD2_0
+                    if (_hostingEnvironment.EnvironmentName.Equals("Production"))
```

Comment:
Should we consider case sensitivity here? Or maybe use `StringComparison.OrdinalIgnoreCase`?

Reply:
I did case sensitivity to reduce cycles. Why set the value if it's already the exact same?